R/FilterState.R

```diff
@@ -267,16 +268,27 @@ FilterState <- R6::R6Class( # nolint
             }
           )
 
+          private$observers[[session$ns("remove")]] <- observeEvent(
+            once = TRUE, # remove button can be called once, should be destroyed afterwards
+            ignoreInit = TRUE, # ignoreInit: should not matter because we destroy the previous input set of the UI
+            eventExpr = input$remove, # when remove button is clicked in the FilterState ui
+            handlerExpr = remove_callback()
+          )
+
           private$destroy_shiny <- function() {
-            logger::log_trace("Destroying FilterState inputs and observers; id: { private$get_id() }")
+            logger::log_debug("Destroying FilterState inputs and observers; id: { private$get_id() }")
+
+            if (session$isEnded()) {
+              return(NULL)
+            } # skip input removal if session has ended
```

Comment:
Finalize error is fixed by ignoring `session$input` cleanup when `session` has ended.

We should not try to clean this ourselves outside of a `teal` session, where we need to make sure the input is removed.

Same fix applied to `FilterStateExpr` that doesn't inherit from this class